src/bootstrap/run.rs

```diff
@@ -41,3 +41,27 @@ fn try_run(builder: &Builder<'_>, cmd: &mut Command) -> bool {
     }
     true
 }
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct InstallGitHook;
+
+impl Step for InstallGitHook {
+    type Output = ();
+
+    /// Runs the `install-git-hook` tool.
+    ///
+    /// This tool in `src/tools` installs a git hook to automatically run
+    /// `tidy --bless` before each commit, so you don't forget to do it
+    fn run(self, builder: &Builder<'_>) {
+        builder.info("Installing git hook");
+        try_run(builder, &mut builder.tool_cmd(Tool::InstallGitHook).arg(&builder.src));
```

Comment:
Hmm, this is small enough it seems like you should be able to add it inline without needing a separate cargo package.
```suggestion
    let script_path: PathBuf = builder.src.join("src/tools/install-git-hook/src/pre-commit.sh");
    let hook_path: PathBuf = builder.src.join(".git/hooks/pre-commit");

    fs::copy(&script_path, &hook_path).expect(
        format!("Failed to copy pre-commit script to {}", &hook_path.to_string_lossy()).as_str(),
    );
```
Not sure where to put `pre-commit.sh` though, maybe in `src/etc`?

Reply:
It could also potentially be inlined as a multiline string and then we could use `fs::write` instead of `fs::copy`, although then we'd have to mess with `chmod`'ing the file as well.

Reply:
I'd rather not inline it so contributors can still run `ln -s` without having to go through x.py. It also makes the shell script easier to edit since you have syntax highlighting and such.

Reply:
Needs to be separate so it can be found by scanning files.